Source/DataSources/KmlDataSource.js

```diff
@@ -1474,27 +1635,298 @@ define([
+                            minRefreshPeriod = defaultValue(queryNumericValue(networkLinkControl, 'minRefreshPeriod', namespaces.kml), 0);
+                        }
+
+                        if (refreshMode === 'onInterval') {
+                            if (hasNetworkLinkControl) {
+                                refreshInterval = Math.max(minRefreshPeriod, refreshInterval);
+                            }
+                            networkLinkInfo.refreshMode = RefreshMode.INTERVAL;
+                            networkLinkInfo.time = refreshInterval;
+                        } else if (refreshMode === 'onExpire') {
+                            var expires;
+                            if (hasNetworkLinkControl) {
+                                expires = queryStringValue(networkLinkControl, 'expires', namespaces.kml);
+                            }
+                            if (defined(expires)) {
+                                try {
+                                    var date = JulianDate.fromIso8601(expires);
+                                    var diff = JulianDate.secondsDifference(date, now);
+                                    if (diff > 0 && diff < minRefreshPeriod) {
+                                        JulianDate.addSeconds(now, minRefreshPeriod, date);
+                                    }
+                                    networkLinkInfo.refreshMode = RefreshMode.EXPIRE;
+                                    networkLinkInfo.time = date;
+                                } catch (e) {
+                                    console.log('KML - NetworkLinkControl expires is not a valid date');
+                                }
+                            } else {
+                                console.log('KML - refreshMode of onExpire requires the NetworkLinkControl to have an expires element');
+                            }
+                        } else {
+                            if (dataSource._camera) { // Only allow onStop refreshes if we have a camera
+                                networkLinkInfo.refreshMode = RefreshMode.STOP;
+                                networkLinkInfo.time = defaultValue(queryNumericValue(link, 'viewRefreshTime', namespaces.kml), 0);
+                            } else {
+                                console.log('A NetworkLink with viewRefreshMode=onStop requires a camera be passed in when creating the KmlDataSource');
+                            }
+                        }
+
+                        if (defined(networkLinkInfo.refreshMode)) {
+                            dataSource._networkLinks.set(networkLinkInfo.id, networkLinkInfo);
+                        }
+                    } else if (viewRefreshMode === 'onRegion'){
+                        console.log('KML - Unsupported viewRefreshMode: onRegion');
                     }
                 });
 
-                dataSource._promises.push(promise);
+                if (defined(dataSource._promises)) {
```

Comment:
Does this check need to be here because `_promises` is only defined during the initial load?  If so, can you add a comment indicating that.  This code looks odd out of context otherwise.